### PR TITLE
chore: bump Go to 1.22

### PR DESCRIPTION
**Pull request by troutowicz**

### CODE REVIEW

**Comment by adrianosela** [comment]
Hey @cosmtrek - a new release supporting the latest version of go would be reaallyyy nice. Air is preventing my team from bumping our projects to 1.22 (since it breaks our engineers' local dev worklows).

**Comment by cosmtrek** [comment]
@troutowicz Hi, could you fix the tests? Thanks

**Comment by troutowicz** [comment]
Hey @cosmtrek, removed that second commit. Will address GH action config another time.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/cosmtrek/air
 
-go 1.21
+go 1.22.0
```
Comment (adrianosela) [inline go.mod:3]: Probably want to keep this on the major version?

Comment (troutowicz) [reply go.mod:3]: Initially tried this but received the following error on `go mod tidy`

```
go: downloading go1.22 (darwin/arm64)
go: download go1.22 for darwin/arm64: toolchain not available
```

Comment (troutowicz) [reply go.mod:3]: https://github.com/golang/go/issues/62278#issuecomment-1693538776

Comment (troutowicz) [reply go.mod]: Fixed. Was a local installation issue.

**Comment by troutowicz** [comment]
> Hey @cosmtrek - a new release supporting the latest version of go would be reaallyyy nice. Air is preventing my team from bumping our projects to 1.22 (since it breaks our engineers' local dev worklows).

We were having the same issue. However, for future blocks, I think setting a newer toolchain will address local dev workflows.

```
go 1.21

toolchain go1.22.0
```

**Pull request merged**